### ISSUE TITLE
Add option to change rendering GPU device and OpenGL engine

### ISSUE THREAD

**Issue by RaptorParkowsky** [colobot/colobot#831]
Also as an option in Device section in the main Settings menu.

I also believe there was an issue about switchable option for OpenGL engine version, but couldn't find it.

**Comment by krzys-h** [comment]
The rendering engine setting is part of issue #505.

What do you mean by changing the rendering GPU? I don't think SDL gives us any control over that.

**Comment by tomangelo2** [comment]
There are some extensions like https://www.opengl.org/registry/specs/NV/gpu_affinity.txt and https://www.opengl.org/registry/specs/AMD/wgl_gpu_association.txt, but I can't tell how it works.

**Comment by RaptorParkowsky** [comment]
By changing the GPU I mean that the most of the games have in their settings panel option to choose the rendering device (for example between integrated Intel and dedicated Nvidia) if they're both installed, enabled and have their drivers also indivudually installed.

Here's some examples to show clearly what I have in mind (sorry for Polish UI descriptions, I couldn't find out how to run those config apps in English):
![schowek01](https://cloud.githubusercontent.com/assets/4019345/18404933/21ee7686-76ee-11e6-934b-475e60669815.png)
![schowek02](https://cloud.githubusercontent.com/assets/4019345/18404936/29134c8e-76ee-11e6-945f-fc2e6d308dd8.png)

I don't know if this is somehow related only to DirectX games or this is Windows-only feature, but I think if this is possible in SDL/OpenGL then it would be good to implement such feature. If the game have already access to default (by OS) rendering device, then I think there should be no problem of choosing other one, right?

Personally for me it would be simply helpful in testing the game on both Intel and Nvidia GPUs almost simultaneously. Theoretically I would even run two separate Colobot: Gold Edition instances every on different GPU without losing so much of performance.

Oh, I've just noticed that for sound output there also could be such option if more than one sound devices are installed in the system...

**Comment by MatiRg** [comment]
We can add option to config about selecting OpenGL Driver from 4 options:
"Auto" - Default will select best aviable OpenGL driver
"1.4" - OGL1.4 Version
"2.1" - OGL2.1 Version
"3.3" - OGL3.3 Version
Option will be save as "OpenGL_Driver"
Game will need to be restarted to apply changes.

**Comment by krzys-h** [comment]
See issue #505 for graphics engine selection. It's not as easy as adding a configuration option in the menu as you also need some kind of fallback mechanizm in case the new settings don't work (i.e. your GPU doesn't support some features required for GL33 engine to work).

**Comment by MatiRg** [comment]
Yeah We need to determine that engine can be used on PC if not switch to "lower" engine I will look at code maybe I will figure something up.
